geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -1197,8 +1193,9 @@ private void restartWithSystem(InternalDistributedSystem newSystem, InternalCach
     logger.info("Locator restart: initializing TcpServer");
 
     try {
-      handler.restarting(newSystem, newCache, configurationPersistenceService);
-      server.restarting();
+      restartHandlers.forEach(
```

Comment:
How did this signal all the handlers before...?

Reply:
The old handler variable was a PrimaryHandler, which broadcast the handler.restarting() call.